### ISSUE TITLE
CxbxKrnl - Launching XBE EP is not blocking code execution

### ISSUE THREAD

**Issue by Luca1991** [Cxbx-Reloaded/Cxbx-Reloaded#617]
I was working on issue #574 and after some hours I realized that g_EmuShared was always nullprt. So I started debugging in the hope to find the cause and... belive it or not, calling the XBE EP is not blocking the  code execution (so EmuShared::Cleanup(); and CxbxKrnlTerminateThread(); are called before the right time).
Where the problem reside:
```
void CxbxLaunchXbe(void(*Entry)())
{
	__try
	{
		Entry();
	}
	__except (EmuException(GetExceptionInformation()))
	{
		printf("Emu: WARNING!! Problem with ExceptionFilter\n");
		
	}

}
```

Entry() call the xbe EP, but it is assumed (or at least I think that it is assumed) that the code execution continues only when closing the xbe. Instead it continues anyway.

This issue is currently a roadblock for issue #574 and might possibly affect some games too.

**Comment by LukeUsher** [comment]
This happens because the the Xbox entry point for most (but not all) titles launches a new thread and then exits. We should probably have a loop underneath this call, which keeps the thread active until the last Xbox thread has terminated

**Comment by x1nixmzeng** [comment]
Bump. This issue is blocking some of my debugger work.

**Comment by x1nixmzeng** [comment]
My workaround is:

```c++

	DbgPrintf("INIT: Calling XBE entry point...\n");
	CxbxLaunchXbe(Entry);

	// FIXME: Wait for Cxbx to exit or error fatally
	Sleep(INFINITE);

	DbgPrintf("INIT: XBE entry point returned\n");
	fflush(stdout);
```

We need `CxbxKrnlInit` to wait for execution to finish - so wait for a fatal error or for Cxbx to exit cleanly.

I wonder what real hardware does in this case. I'm wondering if something isn't implemented correctly when launching threads?

**Comment by x1nixmzeng** [comment]
FYI this workaround was merged along with #909:

https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/blob/c6a031f3fe7826430d54fd6031397834c1428d46/src/CxbxKrnl/CxbxKrnl.cpp#L1124-L1131

**Comment by LukeUsher** [comment]
closing as workaround exists
